ArduPlane/tailsitter.cpp

```diff
@@ -289,15 +289,24 @@ bool QuadPlane::tailsitter_transition_vtol_complete(void) const
 void QuadPlane::tailsitter_check_input(void)
 {
     if (tailsitter_active() &&
+        (plane.control_mode != &plane.mode_qacro) &&
         (tailsitter.input_type & TAILSITTER_INPUT_PLANE)) {
         // the user has asked for body frame controls when tailsitter
-        // is active. We switch around the control_in value for the
+        // is active. We switch around the radio_in value for the
         // channels to do this, as that ensures the value is
         // consistent throughout the code
-        int16_t roll_in = plane.channel_roll->get_control_in();
-        int16_t yaw_in = plane.channel_rudder->get_control_in();
-        plane.channel_roll->set_control_in(yaw_in);
-        plane.channel_rudder->set_control_in(-roll_in);
+        int16_t yaw_in = plane.channel_rudder->get_radio_in();
+        int16_t roll_in = plane.channel_roll->get_radio_in();
+        int16_t roll_trim = plane.channel_roll->get_radio_trim();
+        int16_t roll_delta = roll_in - roll_trim;
+
+        plane.channel_roll->set_radio_in(yaw_in);
+        // reverse the roll input
+        plane.channel_rudder->set_radio_in(roll_trim - roll_delta);
+
+        // now update the control_in values; these are both guaranteed to be TYPE_ANGLE
+        plane.channel_roll->set_control_in(plane.channel_roll->pwm_to_angle());
+        plane.channel_rudder->set_control_in(plane.channel_rudder->pwm_to_angle());
```

Comment:
By using `pwm_to_angle()` are we not using the min max and trim values from the roll channel on a PWM from the yaw channel?

Reply:
good point; maybe it wouldn't be too ugly to fix that though

Reply:
fixed